lib/puppet/indirector/request.rb

```diff
@@ -250,8 +250,6 @@ def set_uri_key(key)
       return
     end
 
-    env, indirector, @key = URI.unescape(uri.path.sub(/^\//, '')).split('/',3)
-    @key ||= ''
-    self.environment = env unless env == ''
+    @key = key
```

Comment:
@joshcooper I think I don't follow after all. The `file` face and resources work because files *are* referenced through URLs and there is a [terminus selector](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_serving/terminus_selector.rb) to make sure the indirector does the right thing, correct? Either way, I don't think we can tackle this through application specific configuration, because of the `agent` use case.

As such, I think we might want to hold on to the selector for `FileBucket::File` after all, so that the same code can keep interacting with local and remote buckets.

The most simple approach might be to just switch to a new and arbitrary URL scheme such as `filebucket://` and move the current catch-all in `Indirector::Request` to a special handling for these.